lib/ocpp/v201/charge_point.cpp

```diff
@@ -814,20 +828,41 @@ void ChargePoint::handle_reset_req(Call<ResetRequest> call) {
     }
 }
 
+void ChargePoint::handle_clear_cache_req(Call<ClearCacheRequest> call) {
+    ClearCacheResponse response;
+    response.status = ClearCacheStatusEnum::Rejected;
+
+    if (this->device_model->get_optional_value<bool>(ControllerComponentVariables::AuthCacheCtrlrEnabled)
+            .value_or(true) and
+        this->database_handler->clear_authorization_cache()) {
+        response.status = ClearCacheStatusEnum::Accepted;
+    }
+
+    ocpp::CallResult<ClearCacheResponse> call_result(response, call.uniqueId);
+    this->send<ClearCacheResponse>(call_result);
+}
+
 void ChargePoint::handle_start_transaction_event_response(CallResult<TransactionEventResponse> call_result,
-                                                          const int32_t evse_id) {
+                                                          const int32_t evse_id, const std::string& id_token) {
     const auto msg = call_result.msg;
-    if (msg.idTokenInfo.has_value() and msg.idTokenInfo.value().status != AuthorizationStatusEnum::Accepted) {
-        if (this->device_model->get_value<bool>(ControllerComponentVariables::StopTxOnInvalidId)) {
-            this->callbacks.stop_transaction_callback(evse_id, ReasonEnum::DeAuthorized);
-        } else {
-            if (this->device_model->get_optional_value<bool>(ControllerComponentVariables::MaxEnergyOnInvalidId)
-                    .has_value()) {
-                // TODO(piet): E05.FR.03
-                // Energy delivery to the EV SHALL be allowed until the amount of energy specified in
-                // MaxEnergyOnInvalidId has been reached.
+    if (msg.idTokenInfo.has_value()) {
+        // C10.FR.05
+        if (this->device_model->get_optional_value<bool>(ControllerComponentVariables::AuthCacheCtrlrEnabled)
+                .value_or(true)) {
+            this->database_handler->insert_auth_cache_entry(utils::sha256(id_token), msg.idTokenInfo.value());
+        }
+        if (msg.idTokenInfo.value().status != AuthorizationStatusEnum::Accepted) {
```

Comment:
Missing C10.FR.03 here, maybe add a todo? Or is it somewhere else?

Reply:
C10.FR.03 is covered by line L216 .